homeassistant/components/technove/coordinator.py

```diff
@@ -16,20 +16,18 @@
 class TechnoVEDataUpdateCoordinator(DataUpdateCoordinator[TechnoVEStation]):
     """Class to manage fetching TechnoVE data from single endpoint."""
 
-    config_entry: ConfigEntry
-
-    def __init__(self, hass: HomeAssistant) -> None:
+    def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
         """Initialize global TechnoVE data updater."""
+        self.technove = TechnoVE(
+            entry.data[CONF_HOST],
+            session=async_get_clientsession(hass),
+        )
         super().__init__(
             hass,
             LOGGER,
             name=DOMAIN,
             update_interval=SCAN_INTERVAL,
         )
-        self.technove = TechnoVE(
-            self.config_entry.data[CONF_HOST],
-            session=async_get_clientsession(hass),
-        )
 
```

Comment:
Why did this change?

Reply:
I've seen a few places that initialise the API object before calling the parent __init__. Now that I can have the entry before the init, I thought I would do the same for consistency and also to fail fast if something is wrong with the TechnoVE object.

Reply:
I mean, what would be wrong with the TechnoVE object? Ideally a constructor should not have side effects like connecting to a device or service.

Reply:
It doesn't harm in this case, IMHO fine to move forward 👍